cmake/Macros.cmake

```diff
@@ -25,17 +25,20 @@ macro(now RESULT)
   endif(WIN32)
 endmacro(now)
 
+find_package(Git)
+
 macro(git_id RESULT)
-  if(WIN32)
-    execute_process(COMMAND "cmd" " /C git --git-dir=${PROJECT_SOURCE_DIR}/.git rev-parse --short=8 HEAD" OUTPUT_VARIABLE ${RESULT})
-    string(REGEX REPLACE "(.*)\n" "\\1" ${RESULT} ${${RESULT}})
-  elseif(UNIX)
-    execute_process(COMMAND "git" "--git-dir=${PROJECT_SOURCE_DIR}/.git" "rev-parse" "--short=8" "HEAD" OUTPUT_VARIABLE ${RESULT})
+  if (GIT_FOUND)
+    if(WIN32)
+      execute_process(COMMAND "cmd" " /C ${GIT_EXECUTABLE} --git-dir=${PROJECT_SOURCE_DIR}/.git rev-parse --short=8 HEAD" OUTPUT_VARIABLE ${RESULT})
+    else()
+      execute_process(COMMAND "${GIT_EXECUTABLE}" "--git-dir=${PROJECT_SOURCE_DIR}/.git" "rev-parse" "--short=8" "HEAD" OUTPUT_VARIABLE ${RESULT})
+    endif()
     string(REGEX REPLACE "(.*)\n" "\\1" ${RESULT} ${${RESULT}})
-  else(WIN32)
-    message(SEND_ERROR "Git ID implemented")
+  else()
+    message(WARNING "Git executable not found, revision number not available.")
     set(${RESULT} 0)
-  endif(WIN32)
+  endif()
```

Comment:
Git will return 128 on a dir without git data
```
git --git-dir=../ rev-parse --short=8 HEAD
fatal: Not a git repository: '../'
echo $?
128
```

you could test for this with code like this:
```
execute_process(COMMAND process param1 RESULT_VARIABLE res_var)
if(NOT "${res_var}" STREQUAL "0")
  # do something here about the failed "process" call...
  message(FATAL_ERROR "process failed res_var='${res_var}'")
endif()
```